scRNA-seq-advanced/01-reading_filtering_CellRanger.Rmd

```diff
@@ -0,0 +1,484 @@
+qcfiltered_sce <- scran::computeSumFactors(qcfiltered_sce, clusters = qclust)
+
+# perform normalization using scaling factors
+# and save as a new SCE object
+normalized_sce <- scuttle::logNormCounts(qcfiltered_sce)
+```
+
+This creates a new "assay" in the `normalized_sce` object, `logcounts`, which contains the normalized count values for each cell and gene.
+
+```{r normalized sce}
+normalized_sce
+```
+
+## Dimension Reduction
+
+Now that we have normalized expression values, we would like to produce some reduced-dimension representations of the data.
+These will allow us to perform some downstream calculations more quickly, reduce some of the noise in the data, and allow us to visualize overall relationships among cells more easily (though with many caveats!).
+
+### Selecting Highly Variable Genes
+
+While we could calculate the reduced dimensions using all of the genes that we have assayed, in practice most of the genes will have very little variation in expression, so doing so will not provide much additional signal.
+Reducing the number of genes we include will also speed up some of the calculations.
+
+To identify the most variable genes, we will use functions from the `scran` package. 
+The first, `modelGeneVar()` attempts to divide the variation we observe for each gene into a biological and technical component, with the intuition that genes with lower mean expression tend to have lower variance for purely technical reasons.
+We then use the output from that with the `getTopHVGs()` function to identify the genes with the highest _biological_ variation, which is  what we are most interested in.
+
+```{r select HVGs}
+# identify 2000 genes 
+num_genes <- 2000
+
+# model variance, partitioning into biological and technical variation
+gene_variance <- scran::modelGeneVar(normalized_sce)
+
+# get the most variable genes
+hv_genes <- scran::getTopHVGs(gene_variance,
+                              n = num_genes)
+```
+
+
+### Principal components analysis
+
+Now that we have selected the genes we would like to use for the reduced-dimension representations of the expression data, we can start to calculate them 
+First we will use the `scater::runPCA()` function to calculate the principal components from the expression matrix. 
+This representation is fast and fairly robust, but may not always provide the best graphical representation, in part because we will usually want to keep a fairly large number of components. 
+
+The default number of components is 50, which we will stick with, but enter it manually just for the record.
+
+```{r runPCA}
+# calculate and save PCA results
+normalized_sce <- scater::runPCA(normalized_sce,
+                                 ncomponents = 50,
+                                 # use only the variable genes we chose
+                                 subset_row = hv_genes)
+```
+
+These reduced dimension results will be stored in a `reducedDim` slot in the SCE object.
```

Comment:
```suggestion
These reduced-dimension results will be stored in a `reducedDim` slot in the SCE object.
```

aha, slot! ;)

Reply:
Yeah, a slot this time... because it is a list(ish), but I don't want to talk about it...